lib/services/role.go

```diff
@@ -2202,7 +2202,7 @@ type KubeResourcesMatcher struct {
 func (m *KubeResourcesMatcher) Match(role types.Role, condition types.RoleConditionType) (bool, error) {
 	var finalResult bool
 	for _, pod := range m.resources {
-		result, err := utils.KubeResourceMatchesRegex(pod, role.GetKubeResources(condition))
+		result, _, err := utils.KubeResourceMatchesRegexWithVerbsCollector(pod, role.GetKubeResources(condition))
```

Comment:
If we are not using the verbs, can we keep using the `KubeResourceMatchesRegex`?

Reply:
`KubeResourceMatchesRegexWithVerbsCollector` ignores the verbs and collects them.
`KubeResourcesMatcher` is used to check if the resources requested via access requests are allowed by the `search_as_roles` rules. We ignore the verbs because we cannot enforce them because we don't know them but they are enforced at runtime.